Content.Server/Medical/CryoPodSystem.cs

```diff
@@ -0,0 +1,367 @@
+                _bloodstreamSystem.TryAddToChemicals(patient.Value, solutionToInject, bloodstream);
+                solutionToInject.DoEntityReaction(patient.Value, ReactionMethod.Injection);
+            }
+        }
+    }
+
+    public void InsertBody(EntityUid uid, EntityUid target, CryoPodComponent cryoPodComponent)
+    {
+        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+            return;
+
+        if (!TryComp<MobStateComponent>(target, out var _comp) || !TryComp<TransformComponent>(target, out var xform))
+            return;
+
+        cryoPodComponent.BodyContainer.Insert(target, transform: xform);
+        xform.LocalPosition = new Vector2(0, 1); // So that the target appears to be floating within the pod
+
+        var comp = EnsureComp<InsideCryoPodComponent>(target);
+        comp.Holder = cryoPodComponent.Owner;
+        _standingStateSystem.Stand(target, force: true); // Force-stand the mob so that the cryo pod sprite overlays it fully
+
+        UpdateAppearance(uid, cryoPodComponent);
+    }
+
+    public void TryEjectBody(EntityUid uid, EntityUid userId, CryoPodComponent? cryoPodComponent)
+    {
+        if (!Resolve(uid, ref cryoPodComponent))
+        {
+            return;
+        }
+
+        if (cryoPodComponent.Locked)
+        {
+            _popupSystem.PopupEntity(Loc.GetString("cryo-pod-locked"), uid, Filter.Entities(userId));
+            return;
+        }
+
+        EjectBody(uid, cryoPodComponent);
+    }
+
+    public void EjectBody(EntityUid uid, CryoPodComponent? cryoPodComponent)
+    {
+        if (!Resolve(uid, ref cryoPodComponent))
+            return;
+
+        if (cryoPodComponent.BodyContainer.ContainedEntity is not {Valid: true} contained)
+            return;
+
+        cryoPodComponent.BodyContainer.Remove(contained);
+        RemComp<InsideCryoPodComponent>(contained);
+        _climbSystem.ForciblySetClimbing(contained, uid);
+
+        // Restore the correct position of the patient. Checking the components manually feels hacky, but I did not find a better way for now.
+        if (TryComp<KnockedDownComponent>(contained, out var _knockedDown)
+            || TryComp<MobStateComponent>(contained, out var mobStateComponent) && _mobStateSystem.IsIncapacitated(contained))
+        {
+            _standingStateSystem.Down(contained);
+        }
+        else
+        {
+            _standingStateSystem.Stand(contained);
+        }
+
+        UpdateAppearance(uid, cryoPodComponent);
+    }
+
+    #region Interaction
+
+    private void HandleDragDropOn(EntityUid uid, CryoPodComponent cryoPodComponent, DragDropEvent args)
+    {
+        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+        {
+            return;
+        }
+
+        var doAfterArgs = new DoAfterEventArgs(args.User, cryoPodComponent.EntryDelay, default, args.Dragged)
```

Comment:
replace the target parameter here (the default you placed down) with the EntityUid of the cryopod, and replace the broadcast finished event with a targeted event instead